src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -240,7 +240,16 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 		// while the UI is not yet ready. Clients have to deal with this fact and we have to make sure that the
 		// stacks model gets updated if any of the UI updating fails with an error.
 		const group = this.ensureGroup(position, !options || !options.preserveFocus);
-		const pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		let pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
```

Comment:
sure, I prefer them to be handled from the outside as well. I wanted to have a constant of options that could be set from a static member.  so it was easy to get opts for `preview`.  I had to set these in a couple places because of editorPicker.

how do you propose I set the `pinned` from the outside? getting access to
`workbench.editor.enablePreviewFromQuickOpen` from the concrete `EditorQuickOpenEntry` was tough.